lib/awful/ewmh.lua

```diff
@@ -246,6 +246,12 @@ local context_mapper = {
 function ewmh.geometry(c, context, hints)
     local layout = c.screen.selected_tag and c.screen.selected_tag.layout or nil
 
+    -- For dockable clients, always apply new geometry
+    if c.dockable and context == "ewmh" then
+        c:geometry(hints.geometry)
+        return
+    end
+
```

Comment:
This wont work as it changes the behavior compared to 4.0 and the next release is a stable one. I propose

```
diff --git a/lib/awful/ewmh.lua b/lib/awful/ewmh.lua
index 817a69bdb..c30fe563d 100644
--- a/lib/awful/ewmh.lua
+++ b/lib/awful/ewmh.lua
@@ -305,10 +305,25 @@ function ewmh.geometry(c, context, hints)
     end
 end
 
+--- Allow the client to move itself.
+--
+-- This is the default geometry request handler when the context is `ewmh`.
+--
+-- @signalhandler awful.ewmh.client_geometry_requests
+-- @tparam client c The client
+-- @tparam string context The context
+-- @tparam[opt={}] table hints The hints to pass to the handler
+function ewmh.client_geometry_requests(c, context, hints)
+    if context == "ewmh" then
+        c:geometry(hints.geometry)
+    end
+end
+
 client.connect_signal("request::activate", ewmh.activate)
 client.connect_signal("request::tag", ewmh.tag)
 client.connect_signal("request::urgent", ewmh.urgent)
 client.connect_signal("request::geometry", ewmh.geometry)
+client.connect_signal("request::geometry", ewmh.client_geometry_requests)
 client.connect_signal("property::border_width", repair_geometry)
 client.connect_signal("property::screen", repair_geometry)
 screen.connect_signal("property::workarea", function(s)
```